main/src/cgeo/geocaching/maps/DirectionDrawer.java

```diff
@@ -0,0 +1,48 @@
+package cgeo.geocaching.maps;
+
+import cgeo.geocaching.location.Geopoint;
+import cgeo.geocaching.maps.interfaces.MapItemFactory;
+import cgeo.geocaching.maps.interfaces.MapProjectionImpl;
+import cgeo.geocaching.settings.Settings;
+
+import android.graphics.Canvas;
+import android.graphics.Paint;
+import android.graphics.Point;
+import android.location.Location;
+
+public class DirectionDrawer {
+    private Geopoint currentCoords;
+    private final Geopoint destinationCoords;
+    private final MapItemFactory mapItemFactory;
+
+    private Paint line = null;
+
+    public DirectionDrawer(final Geopoint coords) {
+        this.destinationCoords = coords;
+        this.mapItemFactory = Settings.getMapProvider().getMapItemFactory();
```

Comment:
This is a bit problematic, as the correct map provider is defined on start of a specific map activity instance. If the DirectionDrawer is created later (again), the settings might have been changed already by a different map activity instance. While this might not be very likely, we have seen this happening in some edge cases in other contexts already. So it would be necessary to hand down the MapItemFactory from the overlay or activity.
If you want I can also add this 'nonfunctional' change.